### DISCUSSION TITLE
How to run desktop mode from node?

### DISCUSSION THREAD

**Discussion by sezganovana** [GoogleChrome/lighthouse#12058] [Q&A] [answered]
Hi!

I'm using node module of LH in my app and I'm struggling to force LH to execute desktop audit. 
LH version is 7.0.1

In LH execution module I've imported desktop config as
```const desktopOpts = require('lighthouse/lighthouse-core/config/desktop-config')```

I also additionally append chrome port and output format into options object
```
lhOptions = desktopOpts
lhOptions.port = chrome.port
lhOptions.output = 'html'
```
My runtime options which are passed to LH are

```
{
  extends: 'lighthouse:default',
  settings: {
    formFactor: 'desktop',      
    throttling: {
      rttMs: 40,
      throughputKbps: 10240,
      cpuSlowdownMultiplier: 1,
      requestLatencyMs: 0,
      downloadThroughputKbps: 0,
      uploadThroughputKbps: 0
    },
    screenEmulation: {
      mobile: false,
      width: 1350,
      height: 940,
      deviceScaleFactor: 1,
      disabled: false
    },
    emulatedUserAgent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) 
AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4143.7 Safari/537.36 Chrome-Lighthouse'
  },
  port: 52522,
  output: 'html'
}
```

### Current behaviour 
Audit is done against mobile device. even though options for desktop device are specified.
it's clearly seen when I run Chrome instance in **non-headless** mode

![mobile audit](https://user-images.githubusercontent.com/74779431/107153812-4abce580-6991-11eb-8166-da5e15ae59f1.png)

### Expected behaviour
Audit to be done against desktop device with the setting passed in options

### Environment
**Node.js**: 12.18.1
**LH**: 7.0.1
**express**: 4.17.1
**Chrome**: 88.0.4324.150 (latest official x64 build)
**OS**: Win 10 Pro, OS build 19041.746

Is there something I'm terribly missing? I tend to think so as Chrome DevTools uses LH as well and when doing audit manually for desktop, it works correctly.

**Comment by patrickhulce** [comment, marked as answer]
Lighthouse options passed in node are not the same thing as the config :)

The signature of Lighthouse 7 is `lighthouse(url, flags, config)`. If you pass the config as the third argument and your extra options as the as the second, I would expect everything to work as expected.

**Reply by sezganovana** [reply]
@patrickhulce thank you so much for explanation. It helped indeed. I assumed that since `flags` is optional parameter, i can only pass the `options`, but didn't notice that flags cannot be passed inside the options. Weird thing though, these flags (`output` and `port`) were processed correctly when passed inside the options object. 

So am I correct in thinking that even though my options config had `extends: 'lighthouse:default'` directive, having the flags inside it made my whole options config to be handled as flags config? 😅

**Reply by patrickhulce** [reply]
> Weird thing though, these flags (output and port) were processed correctly when passed inside the options object.

Output and port *are* options, they're not config properties.

> So am I correct in thinking that even though my options config had extends: 'lighthouse:default' directive, having the flags inside it made my whole options config to be handled as flags config? 😅

I don't know what you mean by an "options config". 

These are your options

```js
{
  port: 52522,
  output: 'html'
}
```

This is your config.

```js
{
  extends: 'lighthouse:default',
  settings: {
    formFactor: 'desktop',      
    throttling: {
      rttMs: 40,
      throughputKbps: 10240,
      cpuSlowdownMultiplier: 1,
      requestLatencyMs: 0,
      downloadThroughputKbps: 0,
      uploadThroughputKbps: 0
    },
    screenEmulation: {
      mobile: false,
      width: 1350,
      height: 940,
      deviceScaleFactor: 1,
      disabled: false
    },
    emulatedUserAgent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) 
AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4143.7 Safari/537.36 Chrome-Lighthouse'
  }
}
```

They shouldn't be merged into 1 object like your initial post. When you pass the merged object as the 2nd argument, the config is ignored and the options are used. When you pass the merged object as the 3rd argument the config is used and the options are ignored.
